backend/siarnaq/api/episodes/models.py

```diff
@@ -338,3 +431,96 @@ class Meta:
 
     def __str__(self):
         return f"{self.tournament} ({self.name})"
+
+    def enqueue(self):
+        """Creates and enqueues all matches for this round.
+        Fails if this round is already in progress."""
+
+        tournament = challonge.get_tournament(self.tournament.challonge_id_private)
+        # Derive matches of this round
+        # NOTE this probably makes more sense (efficiency and consistency)
+        # as a dict. Track in #549
+        matches = []
+        # Takes some wrangling with API response format
+        # We should move this block later
+        # (to keep all code that directly hits challonge
+        # in its own module) Track in #549
+        for item in tournament["included"]:
+            if item["type"] == "match":
+                round_idx = item["attributes"]["round"]
+                if round_idx == self.challonge_id:
+                    # Only enqueue the round if all matches are "open".
+                    # TODO handle ability to requeue a match
+                    # TODO handle ability to requeue an entire round
+                    if item["attributes"]["state"] != "open":
+                        # TODO return some sort of 400, prob 409
+                        # rather than 500-ing
+                        raise Exception
+                    matches.append(item)
+
+        # Map participant "objects" with IDs for easy lookup
+        participants = dict()
+        for item in tournament["included"]:
+            if item["type"] == "participant":
+                id = item["id"]
+                participants[id] = item
+
+        match_objects = []
+        maps_for_match_objects = []
+        match_participant_objects = []
+
+        for m in matches:
+            match_object = Match(
+                episode=self.tournament.episode,
+                tournament_round=self,
+                alternate_order=True,
+                is_ranked=False,
+                challonge_id=m["id"],
+            )
+            match_objects.append(match_object)
+
+            # NOTE these var names suck. Track in #549
+            # or maybe fix earlier if anyone has ideas
+            maps_for_match_objects_batch = [
+                Match.maps.through(match_id=match_object.pk, map_id=map_id)
+                for map_id in self.maps.all()
+            ]
+            maps_for_match_objects += maps_for_match_objects_batch
```

Comment:
I mean, you don't have to construct these in this loop? Can just use a listcomp outside. Something like

```py
maps_for_match_objects = [
    Match.maps.through(...)
    for match in match_objects
    for map_id in self.maps.all()
]
```